src/main/java/net/minecraftforge/common/data/GlobalLootModifierProvider.java

```diff
@@ -87,17 +88,12 @@ public void run(CachedOutput cache) throws IOException
      * Passes in the data needed to create the file without any extra objects.
      *
      * @param modifier      The name of the modifier, which will be the file name.
-     * @param serializer    The registered serializer of this modifier.
      * @param instance      The instance to serialize
      */
-    public <T extends IGlobalLootModifier> void add(String modifier, Codec<T> serializer, T instance)
+    public <T extends IGlobalLootModifier> void add(String modifier, T instance)
     {
-        JsonElement element = serializer.encodeStart(JsonOps.INSTANCE, instance).getOrThrow(false, s -> {});
-        if (!(element instanceof JsonObject obj))
-            throw new RuntimeException("Expecting loot modifier to be a Json Object!");
-        String serializerName = ForgeRegistries.LOOT_MODIFIER_SERIALIZERS.get().getKey(serializer).toString();
-        obj.addProperty("type", serializerName);
-        this.toSerialize.put(modifier, obj);
+        JsonElement json = IGlobalLootModifier.DIRECT_CODEC.encodeStart(JsonOps.INSTANCE, instance).getOrThrow(false, s -> {});
```

Comment:
Vanilla uses change the consumer to `LOGGER::error`. I think u should also do that here

Reply:
Confirmed `getOrThrow` throws an exception already and it gets logged.